fem/ceed/nlconvection_qf.h

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+/// A structure used to pass additional data to f_build_conv and f_apply_conv
+struct NLConvectionContext { CeedInt dim, space_dim, vdim; CeedScalar coeff; };
+
+/// libCEED Q-function for building quadrature data for a convection operator
+/// with a constant coefficient
+CEED_QFUNCTION(f_build_conv_const)(void *ctx, CeedInt Q,
+                                   const CeedScalar *const *in,
+                                   CeedScalar *const *out)
+{
+   NLConvectionContext *bc = (NLConvectionContext*)ctx;
+   // in[0] is Jacobians with shape [dim, nc=dim, Q]
+   // in[1] is quadrature weights, size (Q)
+   //
+   // At every quadrature point, compute qw/det(J).adj(J).adj(J)^T and store
+   // the symmetric part of the result.
```

Comment:
This comment and other comments below seem to be copy-pasted from another place and not updated -- they are incorrect for this integrator I think.